google/cloud/storage/_signing.py

```diff
@@ -586,8 +586,7 @@ def generate_signed_url_v4(
     if generation is not None:
         query_parameters["generation"] = generation
 
-    ordered_query_parameters = sorted(query_parameters.items())
-    canonical_query_string = six.moves.urllib.parse.urlencode(ordered_query_parameters)
```

Comment:
No more standard Python `urlencode()`

Reply:
I don't understand that comment:

```bash
$ .nox/unit-3-8/bin/python
Python 3.8.1 (default, Feb 11 2020, 17:29:15) 
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
>>> six.moves.urllib
<module 'six.moves.urllib' (<six._SixMetaPathImporter object at 0x7fc1d4781550>)>
>>> six.moves.urllib.parse
<module 'six.moves.urllib_parse'>
>>> six.moves.urllib.parse.urlencode
<function urlencode at 0x7fc1d3e59b80>
```

Reply:
> OK, I see now that you mean to say we can't use urlencode, because it doesn't quote according to the V4 spec.
Yes

Reply:
OK, so we can't use `urlencode`:  is there a reason why we can't use the stdlib's `quote` function, passing our own `safe` set?

Reply:
I think the implementation LGTM. @tseaver do you still have concerns?